### PR TITLE
Fixes smart pipe & cryo gas reallocation

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request

**1. Smart Pipe Problem**
 - Fill a pipe with any gas, plasma for e.g.
 - Unwrench the pipe to deconstruct
 - The pressure sends you flying but notice plasma is nowhere to be seen from the now destroyed pipe
 
**2. Cryo pipe connector problem**
The gas gets reallocated when the object is deleted. We only want this when its deconstructed not deleted to prevent any side effects

This PR fixes them

## Changelog
:cl:
fix: smart pipes release their gases into the air when unwrenched
fix: cryo pipe connector component has no side effects of gas reallocation when deleted
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
> Looks like I did not do a good enough job, did not notice that qdel sign up, I am sorry. 😔

Shit happens :)

**Comment by SyncIt21** [comment]
> Something isn't right with this. Unwrenching a pipe releases all of the gas in the entire network, but removes the correct amount from the pipes, seemingly duplicating gas

It seems return_air() gives us parent.air which is the pipenet shared by all pipes.

Need to find the specific air for this pipe only. I'll look into it

**Comment by Kapu1178** [comment]
> > Something isn't right with this. Unwrenching a pipe releases all of the gas in the entire network, but removes the correct amount from the pipes, seemingly duplicating gas
> 
> It seems return_air() gives us parent.air which is the pipenet shared by all pipes.
> 
> Need to find the specific air for this pipe only. I'll look into it tomorrow

return_air() returns the parent pipe`line`, iirc. A pipeline a group of pipes seperated by something such as a valve or pump.

**Comment by SyncIt21** [comment]
> Something isn't right with this. Unwrenching a pipe releases all of the gas in the entire network, but removes the correct amount from the pipes, seemingly duplicating gas

Right turns out we need to first destory the pipenet for it to allocate `air_temporary` to our pipes and only then can we release it
https://github.com/tgstation/tgstation/blob/df3de65e62ed01d3f9408534329c9a545fc7d1da/code/modules/atmospherics/machinery/datum_pipeline.dm#L35-L36

Which does this
https://github.com/tgstation/tgstation/blob/df3de65e62ed01d3f9408534329c9a545fc7d1da/code/modules/atmospherics/machinery/datum_pipeline.dm#L207

This was always done for pipes before it was broken by #82659. Because `on_deconstruction()` gets called before `Destroy()` we tried to use `air_temporary` before it could be initialized by `datum/pipeline/Destroy()` so no air was initialized to release.

@FlufflesTheDog can you retest now to confirm it works?

**Comment by FlufflesTheDog** [comment]
Yeah, looks good 👍

File: code/modules/atmospherics/machinery/components/unary_devices/machine_connector.dm [hunk 1]
```diff
@@ -41,7 +41,7 @@
 	RegisterSignal(connected_machine, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(pre_move_connected_machine))
 	RegisterSignal(connected_machine, COMSIG_MOVABLE_MOVED, PROC_REF(moved_connected_machine))
 	RegisterSignal(connected_machine, COMSIG_MACHINERY_DEFAULT_ROTATE_WRENCH, PROC_REF(wrenched_connected_machine))
-	RegisterSignal(connected_machine, COMSIG_QDELETING, PROC_REF(deconstruct_connected_machine))
+	RegisterSignal(connected_machine, COMSIG_OBJ_DECONSTRUCT, PROC_REF(deconstruct_connected_machine))
```
Comment (jlsnow301) [inline code/modules/atmospherics/machinery/components/unary_devices/machine_connector.dm:44]: imo, qdeleting should just call its own handler which performs relocate_airs() and such (basically the same event handler but without qdel'ing itself)

deconstruct handler should then unregister qdel before qdeling

Comment (SyncIt21) [reply code/modules/atmospherics/machinery/components/unary_devices/machine_connector.dm:44]: > qdeleting should just call its own handler which performs relocate_airs()

We want relocation of airs only upon deconstruction and not deletion else that would be a side effect.

But you right. Both deletion and deconstruction share common code except for relocate_airs().

I'll fix that 👍

**Pull request merged**